File: wavesplatform/src/main/java/com/wavesplatform/sdk/model/request/ReissueTransactionRequest.java
package com.wavesplatform.sdk.model.request;

import android.util.Log;

import com.google.common.primitives.Bytes;
import com.google.common.primitives.Longs;
import com.wavesplatform.sdk.crypto.Base58;
import com.wavesplatform.sdk.crypto.CryptoProvider;

public class ReissueTransactionRequest {
    public static int MinFee = 100000000;

    final public String assetId;
    final public String senderPublicKey;
    final public long quantity;
    final public boolean reissuable;
    final public long fee;
    final public long timestamp;
    public String signature;

    public transient final int txType = 5;

    public ReissueTransactionRequest(String assetId, String senderPublicKey, long quantity,
                                     boolean reissuable, long timestamp) {
        this.assetId = assetId;
        this.senderPublicKey = senderPublicKey;
        this.quantity = quantity;
        this.reissuable = reissuable;
        this.fee = MinFee;
        this.timestamp = timestamp;
    }

    public byte[] toSignBytes() {
        try {
            byte[] reissuableBytes = reissuable ? new byte[]{1} : new byte[]{0};

            return Bytes.concat(new byte[]{txType},
                    Base58.decode(senderPublicKey),
                    Base58.decode(assetId),
                    Longs.toByteArray(quantity),
                    reissuableBytes,
                    Longs.toByteArray(fee),
                    Longs.toByteArray(timestamp));
        } catch (Exception e) {
            Log.e("ReissueRequest", "Couldn't create toSignBytes", e);
            return new byte[0];
        }
    }

    public void sign(byte[] privateKey) {
        if (signature == null) {
            signature = Base58.encode(CryptoProvider.sign(privateKey, toSignBytes()));
        }
    }
}

File: wavesplatform/src/main/java/com/wavesplatform/sdk/net/model/request/ReissueTransactionRequest.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.sdk.net.model.request

import android.util.Log

import com.google.common.primitives.Bytes
import com.google.common.primitives.Longs
import com.wavesplatform.sdk.crypto.Base58
import com.wavesplatform.sdk.crypto.CryptoProvider

class ReissueTransactionRequest(val assetId: String, val senderPublicKey: String, val quantity: Long,
                                val reissuable: Boolean, val timestamp: Long) {
    val fee: Long
    var signature: String? = null

    @Transient
    val txType = 5

    init {
        this.fee = MIN_FEE
    }

    fun toSignBytes(): ByteArray {
        try {
            val reissuableBytes = if (reissuable) byteArrayOf(1) else byteArrayOf(0)

            return Bytes.concat(byteArrayOf(txType.toByte()),
                    Base58.decode(senderPublicKey),
                    Base58.decode(assetId),
                    Longs.toByteArray(quantity),
                    reissuableBytes,
                    Longs.toByteArray(fee),
                    Longs.toByteArray(timestamp))
        } catch (e: Exception) {
            Log.e("ReissueRequest", "Couldn't create toSignBytes", e)
            return ByteArray(0)
        }

    }

    fun sign(privateKey: ByteArray) {
        if (signature == null) {
            signature = Base58.encode(CryptoProvider.sign(privateKey, toSignBytes()))
        }
    }

    companion object {
        const val MIN_FEE = 100000000L
    }
}
